loginPlugins/beaconing.js

```
@@ -0,0 +1,329 @@
+ *  data usefull for the login (username, email, access control token).
+ * @type {string}
+ */
+var callback = 'https://localhost:3000/api/proxy/afront/loginbyplugin';
+
+/**
+ * The 'plugin id'. Unique identifier of the plugin that is used to identify it.
+ * It's used to set up the route of this plugin, e.g. '/login/pluginId' -> (pluginId = samlnl) -> '/login/samlnl'
+ *
+ * @type {string}
+ */
+var pluginId = 'beaconing';
+
+/**
+ * Plugin name.
+ * @type {string}
+ */
+var pluginName = 'beaconing';
+
+/**
+ * Information for the passport configuration.
+ * @type {{path: string, entryPoint: string, issuer: string}}
+ */
+
+var passport = {
+    // ?response_type=code&client_id=some_client_id_here&redirect_uri=some_callback_url_here
+    baseURL: 'https://core.beaconing.eu/',
+    authorizationURL: 'https://core.beaconing.eu/auth/auth',
+    tokenURL: 'https://core.beaconing.eu/auth/token',
+    clientID: 'analytics',
+    clientSecret: 'IVzeBVKpPriNYbM3O4Dr4El01R9zTBkNsgrAahvoiu7f51oUws3ISIAJOkBhlzuA',
+    callbackURL: '',
+    partialURL: '/api/login/beaconing/callback'
+};
+
+function oauthSetup(app) {
+
+    var getUrl = function(req){
+        return req.protocol + '://' + req.get('host');
+    };
+    
+    var hasKey = function(array, key){
+        var has = false;
+
+        for (var i = array.length - 1; i >= 0; i--) {
+            if(array[i].domain === key){
+                has = true;
+                break;
+            }
+        }
+
+        return has;
+    };
+
+    var checkUser = function(accessToken, refreshToken, profile, done) {
+        console.log(profile);
+
+        userExists(profile, app.db, function (err, user) {
+            if (err) {
+                // No user found, create a new user and assign a new role
+                console.info(err);
+                return addNewUser(profile, app.db, app.acl, done);
+            }
+
+            var userRoles = function(){
```

If the user exist you should return an error. Imagine that the user already exist, these user maybe is not the same that the new user that you are registering.

---

I've made it configurable with a flag. By default it'll return an error!
Nice catch